### ISSUE TITLE
Documentation for blocks

### ISSUE THREAD

**Issue by SgtOddball** [madebymany/sir-trevor-js#451]
I know it's been mentioned before but after having a custom block thats been 'almost' working flat out explode on me this afternoon I was wondering when there's going to be a run down on how to make your own custom block for Sir Trevor. 

A good example would be adding in the current 'code' block... it flat out doesn't work out of the box.

Add in the global, factory etc. function + "use strict" to at least correctly initialise the block and you get an icon..(still doesn't work but it's still in a better position than it was). 

Long story short, any chance of something that tells you exactly what you need and what the available options are without a massive amount of flailing around hoping to find the magical trigger that fires the block back up again?

**Comment by SgtOddball** [comment]
<rant>
Oh and I finally found out what caused it to die on it's arse.... having a second capital letter in the block name.....

I mean....really. Why? Why would it matter so much if there's more than one so long as it's consistent? 
</rant>

**Comment by raffij** [comment]
Hi @SgtOddball,

Nothing wrong with a good rant.

Public API and developer docs are something we've been discussing internally.

Been waiting to modify them before creating proper docs, but that's a while off, so makes sense I write some for the state now to make it easier. Also remake the example blocks so they all work as required, as it's also likely when they do work don't work in the best way.

Raffi

**Comment by SgtOddball** [comment]
It was one of those 'IT WAS WORKING!?!!!' kind of moments before spending an hour trying to figure out why 'BlockLink' kept being spat out.

**Comment by raffij** [comment]
You saying this one isn't working?

https://github.com/neyre/sir-trevor-wp/blob/master/custom-blocks/Code.js

**Comment by SgtOddball** [comment]
Yeah, I couldn't get it to work, but after an evening coding session I've finally nailed got it behaving it's self.

When I've got the time I'll stick a repository up of it but it's just to wrap a block up in a link so that the whole block can become a link/function/what ever when it comes out the other end.

But anyways.. I'll admit it was ranting abit earlier but the capital letter thing was one of those bugs thats good enough to drive a dev to tears in the server room.

**Comment by SgtOddball** [comment]
Oh and note to self... loadData happens before onRender and if you don't want it to drop the data and reload it again, trigger this.ready.... yeah that one was fun.

**Comment by raffij** [comment]
All great feedback for thinking about the future stable Apis.

**Comment by raffij** [comment]
Documenting the current block api today. We will then be working towards a better defined api in the near future.

**Comment by SgtOddball** [comment]
Cool, well I've calmed down since this happened (I've slept at least a couple of times.... :grinning: ) but I'd be interested to see how the new info lines up with what I don' learned...

**Comment by raffij** [comment]
@SgtOddball i've added some basic flows to the docs to help with this for now http://madebymany.github.io/sir-trevor-js/docs.html#4-1

**Comment by SgtOddball** [comment]
I shall get time to read this at some point this week (so much for having some time to go through this stuff over the last fortnight).
